### PR TITLE
Extractor2

### PR DESCRIPTION
**Pull request by Christian-B**
prototype 2

Needs Donal's magic

### CODE REVIEW

File: SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/DataGatherRunner.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        MachineBean fromJson = mapper.readValue(
+                machineReader, MachineBean.class);
+        Machine machine = new Machine(fromJson);
+
+        Transceiver trans = new Transceiver(
+                machine.getBootEthernetAddress(), machine.version);
+
+        String database = args[THIRD];
+
+        //TODO Donal's magic!
```
Comment (dkfellows) [inline SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/DataGatherRunner.java]: Hmm!

File: SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/Placement.java [hunk 2]
```diff
@@ -38,11 +43,16 @@
     Placement(@JsonProperty(value = "x", required = true) int x,
             @JsonProperty(value = "y", required = true) int y,
             @JsonProperty(value = "p", required = true) int p,
-            @JsonProperty(value = "vertex", required = true) Vertex vertex) {
+            @JsonProperty(value = "vertex", required = true) Vertex vertex,
+            @JsonProperty(value = "iptag", required = false) IPTag iptag,
```
Comment (dkfellows) [inline SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/Placement.java]: Is this right?

File: SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/DataGatherRunner.java (hunk 1 above)
Comment (Christian-B) [reply SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/DataGatherRunner.java]: Needs more magic than that!

File: SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/Placement.java (hunk 2 above)
Comment (Christian-B) [reply SpiNNaker-front-end/src/main/java/uk/ac/manchester/spinnaker/front_end/interfaces/buffer_management/Placement.java]: NOPE!

**Comment by Christian-B** [comment]
java.lang.IllegalArgumentException: The tag port must have been set
	at uk.ac.manchester.spinnaker.transceiver.Transceiver.setIPTag(Transceiver.java:1667)
	at uk.ac.manchester.spinnaker.front_end.interfaces.buffer_management.DataGatherer.reconfigureIPtag(DataGatherer.java:229)
	at uk.ac.manchester.spinnaker.front_end.interfaces.buffer_management.DataGatherer.downloadBoard(DataGatherer.java:203)
	at uk.ac.manchester.spinnaker.front_end.interfaces.buffer_management.DataGatherer.lambda$addTask$0(DataGatherer.java:104)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by Christian-B** [comment]
DirectDataGatherer.getCoreRegionTable are you looking for the maximum size of the recording region or the size used in this run? Ie for spike do you want the size the spikes could occupy or just the size of the current spike data?

**Comment by dkfellows** [comment]
For the `DirectDataGatherer`, there's no difference. _It doesn't (and can't) handle recording regions._ That's why there's a `RecordingRegionDataGatherer` too.

**Comment by dkfellows** [comment]
Depends on #70, which merges to this branch.

**Comment by dkfellows** [comment]
Depends on having #210 done

**Comment by dkfellows** [comment]
<h2>IT LIVES!</h2>
Bwahahahaaa!

**Comment by Christian-B** [comment]
Agreed early testing is successful.
Currently to test this you have to.
1. Use the extractor2 branch of DS, PACMAN, FEC and Spynakker (and optionally GFE)
2. Set the [Java] values in .spynnaker.cfg
[Java]
use_java = False
# call to start a java process.
# If there is no jdk../bin in your class path this must include the full path
java_call = java
# Absolute path to where the JavaSpiNNaker git repository is located.
# If none it will be assumed to be in the same parent directory as SpiNNFrontEndCommon
java_spinnaker_path = None

**Comment by Christian-B** [comment]
This is the main branch so not calling it master is silly.

**Pull request merged**